server/src/main/java/org/elasticsearch/common/lucene/search/XMoreLikeThis.java

```diff
@@ -910,9 +912,9 @@ private void addTermFrequencies(Reader r, Map<String, Int> termFreqMap, String f
                 // increment frequency
                 Int cnt = termFreqMap.get(word);
                 if (cnt == null) {
-                    termFreqMap.put(word, new Int());
+                    termFreqMap.put(word, new Int(tfAtt.getTermFrequency()));
```

Comment:
If we merge this PR when we start using the modified Lucene MoreLikeThis class, we also need to adapt this in a similar location in `addTermFrequencies(Map<String, Int>, Terms , @Nullable String)` (L870 and before). That code path is used in the `like(Fields... likeFields)` method which is part of the code XMoreLikeThis contains thats not in the Lucene twin class.